lalrpop/src/grammar/repr.rs

```diff
@@ -222,13 +222,13 @@ impl TypeRepr {
                               None => vec![],
                           })
                           .collect(),
-            TypeRepr::Associated { type_parameter, .. } =>
-                vec![TypeParameter::Id(type_parameter)],
-            TypeRepr::Lifetime(l) =>
-                vec![TypeParameter::Lifetime(l)],
+            TypeRepr::Associated { ref type_parameter, .. } =>
+                vec![TypeParameter::Id(type_parameter.clone())],
```

Comment:
sad that `Atom` is not copy; but what you gonna' do

Reply:
I think I agree with it not being copy. I'd probably have more conviction if I hadn't just changed all that code. :)

Reply:
Well, the reason is that they maintain a reference count. Seems fine.